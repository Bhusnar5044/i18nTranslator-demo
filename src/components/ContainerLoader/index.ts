export { ContainerLoader as default } from './ContainerLoader';
