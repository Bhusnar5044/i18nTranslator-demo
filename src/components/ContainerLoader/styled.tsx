import { keyframes, styled } from '@utils';

const logoSpin = keyframes`
   from {
            transform: rotate(0deg);
        }
        to {
            transform: rotate(360deg);
        }
`;
export const Loader = styled.img`
    width: 10rem;
    height: auto;
    margin: auto;
    animation: ${logoSpin} infinite 20s linear;
`;
