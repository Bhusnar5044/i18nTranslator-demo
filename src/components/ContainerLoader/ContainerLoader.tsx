import { FC, memo } from 'react';
import { Loader } from './styled';

export const ContainerLoader: FC = memo(() => {
    return <Loader src="/reactLogo.svg" title="loader" />;
});

ContainerLoader.displayName = 'ContainerLoader';
