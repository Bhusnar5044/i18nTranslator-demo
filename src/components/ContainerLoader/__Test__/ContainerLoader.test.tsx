import { render, screen } from '@test-utils';
import { ContainerLoader } from '../ContainerLoader';

describe('ContainerLoader', () => {
    it('should render properly', () => {
        const { container } = render(<ContainerLoader />);
        expect(container).toMatchSnapshot();
        expect(screen.getByTitle('loader')).toBeInTheDocument();
    });
});
