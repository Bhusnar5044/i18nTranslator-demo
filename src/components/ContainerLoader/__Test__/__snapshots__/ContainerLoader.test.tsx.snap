// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerLoader > should render properly 1`] = `
<div>
  <img
    class="sc-eLqIir iGbLNM"
    src="/reactLogo.svg"
  />
</div>
`;

exports[`ContainerLoader should render properly 1`] = `
<div>
  <img
    class="sc-eLqIir iGbLNM"
    src="/reactLogo.svg"
    title="loader"
  />
</div>
`;
